cert/cert.go

```diff
@@ -398,34 +434,34 @@ func (spec *Spec) Lifespan() time.Duration {
 	}
 
 	// This bit of code is necessary to confirm that the cert/key are older than the spec definition.
+	if spec.IsChangedOnDisk(spec.Key.Path) || spec.IsChangedOnDisk(spec.Cert.Path) {
+		// This is necessary to essentially force cfssl to regenerate since it's not spec aware.
+		log.Infof("refreshing due to spec %s having a newer mtime than key or cert", spec.Path)
+		spec.ResetLifespan()
+		return 0
+	}
+	return spec.tr.Lifespan()
+}
+
+func (spec *Spec) IsChangedOnDisk(path string) bool {
 	specStat, err := os.Stat(spec.Path)
 	if err != nil {
 		// The assertion here is that the spec actually
 		// exists. If it doesn't, something is wrong with the
 		// world.
-		panic("cert: certificate spec doesn't exist during RefreshKeys()")
+		panic("cert: isChangedOnDisk: failed to stat certificate spec")
 	}
-
-	isTooOld := func(path string) bool {
-		st, err := os.Stat(path)
-		if err != nil {
-			if os.IsNotExist(err) {
-				log.Errorf("while checking cert/key path %s, got path error %s", path, err)
-			}
-			return true
-		}
-		if specStat.ModTime().After(st.ModTime()) {
-			log.Infof("refreshing due to spec %s having a newer mtime then %s", spec.Path, path)
-			return true
+	st, err := os.Stat(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			log.Errorf("cert isChangedOnDisk: while checking path %s, got path error %s", path, err)
 		}
-		return false
+		return true
 	}
-	if isTooOld(spec.Key.Path) || isTooOld(spec.Cert.Path) {
-		// This is necessary to essentially force cfssl to regenerate since it's not spec aware.
-		spec.ResetLifespan()
-		return 0
+	if specStat.ModTime().After(st.ModTime()) {
```

Comment:
could just do
`return specStat.ModTime().After(st.ModTime())`
And collapse out an if block.  Not a hard requirement, just an observation-

Reply:
Good catch! Thanks :)